### PR TITLE
feat(discover): Show banner for split decision

### PR DESCRIPTION
**Pull request by shruthilayaj**
1. Show banner for split decision
2. Use saved query dataset to decide which actual dataset to query
3. Switching datasets preserves old state and only switches dataset instead of resetting the whole query when already in a saved query - it's a better experience if we guess the wrong dataset, the state remains intact as the switch between datasets
4. Removes references to queryDataset in EventView to keep event dataset selector agnostic

### CODE REVIEW

File: static/app/views/discover/eventDetails/index.spec.tsx [hunk 1]
```diff
@@ -212,19 +212,19 @@ describe('Discover > EventDetails', function () {
     expect(await screen.findByText('Firefox')).toBeInTheDocument();
     expect(screen.getByRole('link', {name: 'Firefox'})).toHaveAttribute(
       'href',
-      '/organizations/org-slug/discover/results/?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All%20Events&query=browser%3AFirefox%20title%3A%22Oh%20no%20something%20bad%22&queryDataset=discover&sort=-timestamp&statsPeriod=24h&yAxis=count%28%29'
+      '/organizations/org-slug/discover/results/?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All%20Events&query=browser%3AFirefox%20title%3A%22Oh%20no%20something%20bad%22&sort=-timestamp&statsPeriod=24h&yAxis=count%28%29'
```
Comment (shruthilayaj) [inline static/app/views/discover/eventDetails/index.spec.tsx:215]: No longer passing the `queryDataset=discover` param if the feature flag isn't enabled

File: static/app/utils/discover/eventView.tsx [hunk 2]
```diff
@@ -38,12 +38,12 @@ import {
   DISPLAY_MODE_FALLBACK_OPTIONS,
```
Comment (shruthilayaj) [inline static/app/utils/discover/eventView.tsx:38]: The changes in this file is almost entirely removing the `queryDataset` param from EventView. the discover component instead handles figuring which `dataset` value to pass to Eventview.

File: static/app/views/discover/homepage.tsx [hunk 3]
```diff
@@ -108,10 +109,18 @@ class HomepageQueryAPI extends DeprecatedAsyncComponent<Props, HomepageQueryStat
 
   renderBody(): React.ReactNode {
     const {savedQuery, loading} = this.state;
+    const {organization} = this.props;
+    let savedQueryWithDataset = savedQuery;
+    if (
+      organization.features.includes('performance-discover-dataset-selector') &&
+      savedQuery
+    ) {
+      savedQueryWithDataset = getSavedQueryWithDataset(savedQuery);
```
Comment (shruthilayaj) [inline static/app/views/discover/homepage.tsx:118]: infers `dataset` based on saved query dataset. `dataset` is then passed down to event view so the queries are made correctly.

File: static/app/views/discover/results.spec.tsx [hunk 4]
```diff
@@ -648,12 +648,12 @@ describe('Results', function () {
 
       expect(screen.getByRole('link', {name: 'timestamp'})).toHaveAttribute(
         'href',
-        'undefined?field=title&field=event.type&field=project&field=user.display&field=timestamp&id=1&name=new&query=&queryDataset=discover&sort=-timestamp&statsPeriod=24h&topEvents=5'
+        'undefined?field=title&field=event.type&field=project&field=user.display&field=timestamp&id=1&name=new&query=&sort=-timestamp&statsPeriod=24h&topEvents=5'
```
Comment (shruthilayaj) [inline static/app/views/discover/results.spec.tsx:651]: No longer passing the queryDataset=discover param if the feature flag isn't enabled

File: static/app/views/discover/savedQuery/datasetSelector.spec.tsx [hunk 5]
```diff
@@ -1,26 +1,57 @@
...
+      'Transactions',
+      'Unknown',
+    ]);
   });
 
-  it('pushes new event view', async function () {
-    render(<DatasetSelector isHomepage={false} savedQuery={undefined} />, {
-      router,
-    });
+  it('pushes new deafult event view if not a saved query', async function () {
```
Comment (narsaynorath) [inline static/app/views/discover/savedQuery/datasetSelector.spec.tsx:47]:
```suggestion
  it('pushes new default event view if not a saved query', async function () {
```

File: static/app/utils/discover/eventView.tsx [hunk 6]
```diff
@@ -609,6 +599,13 @@ class EventView {
       delete newQuery.query;
     }
 
+    if (this.dataset) {
+      newQuery.queryDataset = getSavedQueryDatasetFromLocationOrDataset(
+        undefined,
+        this.dataset
+      );
+    }
```
Comment (narsaynorath) [inline static/app/utils/discover/eventView.tsx:607]: I'm a little confused about why we need to pass along the `queryDataset` field to a new query if these changes were meant to remove this, could `EventView` just handle a single `dataset` field? Is this being used?

**Review by narsaynorath — APPROVED** [review]
Just a small typo and a comment r.e. `queryDataset` in `EventView` but otherwise looks good

File: static/app/utils/discover/eventView.tsx (hunk 6 above)
Comment (shruthilayaj) [reply static/app/utils/discover/eventView.tsx:607]: I went back and forth on this a bit, `toNewQuery()` is primarily used to create the payload for creating/updating a DiscoverSavedQuery object. So I figured I'd rather implement it here than at every individual discover saved query api call. When we implement this change for dashboards, we could add another helper method to EventView like `toNewWidgetQuery()` and do something similar.

Comment (narsaynorath) [reply static/app/utils/discover/eventView.tsx:607]: I see, yeah, I think if this is supposed to produce the most valid instance of a `DiscoverSavedQuery` object then I'd rather us keep the logic together than having to remember to add the `queryDataset` attribute everywhere it's called. Thanks for clarifying why this is here!

**Pull request merged**